LISA/Executor.cpp

```diff
@@ -471,11 +539,79 @@ void Executor::doUninstall(std::string type, std::string id, std::string version
         Filesystem::removeDirectory(appStoragePath);
     }
 
-    // TODO invoke maintenace and cleanup
+    doMaintanace();
 
     INFO("finished");
 }
 
+void Executor::doMaintanace()
+{
+    try {
+        // clear tmp
+        Filesystem::removeDirectory(config.getAppsTmpPath());
+        Filesystem::createDirectory(config.getAppsTmpPath());
+
+        // remove installed apps data not present in installed_apps
+        auto appsPathRoot = config.getAppsPath() + Filesystem::LISA_EPOCH + '/';
+        auto foundApps = scanDirectories(appsPathRoot);
+        for (const auto& app : foundApps) {
+            INFO(app);
+            if (dataBase->IsAppInstalled(app.type, app.id, app.version)) {
+                ERROR(app, " not found in installed apps, removing dir");
+                auto path = config.getAppsPath() + Filesystem::createAppPath(app.type, app.id, app.version);
+                Filesystem::removeDirectory(path);
+            }
+        }
+
+        // remove apps data not present in apps
+        auto appsStoragePathRoot = config.getAppsStoragePath() + Filesystem::LISA_EPOCH + '/';
+        auto foundAppsStorages = scanDirectories(appsStoragePathRoot);
+        for (const auto& app : foundAppsStorages) {
+            INFO(app);
+            if (dataBase->IsAppData(app.type, app.id)) {
+                ERROR(app, " not found in apps, removing dir");
+                auto path = config.getAppsStoragePath() + Filesystem::createAppPath(app.type, app.id);
+                Filesystem::removeDirectory(path);
+            }
+        }
+
+        auto appsDetailsList = dataBase->GetAppDetailsList();
+        for (const auto& details : appsDetailsList) {
+            INFO("details: ", details.id, ":", details.version);
+
+            auto appPaths = dataBase->GetAppsPaths(details.type, details.id, details.version);
+
+            INFO("PATHS APPS:");
+            for (const auto& path : appPaths) {
+                INFO("path: ", path);
+                auto appPath = config.getAppsPath() + path;
+                INFO("abs path: ", appPath);
+
+                bool noAppFiles = Filesystem::directoryExists(appPath) ? Filesystem::isEmpty(appPath) : true;
+                if (noAppFiles) {
+                    dataBase->RemoveInstalledApp(details.type, details.id, details.version);
+                }
+            }
+
+            auto dataPaths = dataBase->GetDataPaths(details.type, details.id);
+
+            INFO("PATHS DATA:");
+            for (const auto& path : dataPaths) {
+                INFO("path: ", path);
+                auto dataPath = config.getAppsStoragePath() + path;
+                INFO("abs path: ", dataPath);
+                if (Filesystem::directoryExists(dataPath)) {
```

Comment:
I expect you also want to reverse the logic here: if NOT directory exists then create it?